tests/cmd/auth/test_login.py

```diff
@@ -426,3 +452,57 @@ def _wait_for_callback(self, *args, **kwargs):
                     raise ClickException(e.message)
 
         return FakeOAuthClient
+
+    @pytest.mark.parametrize(
+        ["version", "preference_enabled", "status_code", "expected_error"],
+        [
+            ["2022.04.0", True, 200, VERSION_TOO_LOW_MESSAGE],
+            ["v1.0.0", None, 404, VERSION_TOO_LOW_MESSAGE],
+            ["v1.0.0", False, 200, DISABLED_FLOW_MESSAGE],
+            ["v1.0.0", True, 200, None],
+            [
+                "v1.0.0",
+                None,
+                200,
+                None,
+            ],  # Removing the preference is like making it on by default
+        ],
+    )
+    def test_assert_flow_enabled(
+        self,
+        monkeypatch,
+        version,
+        preference_enabled,
+        status_code,
+        expected_error,
+    ):
+        """
+        GIVEN -
+        WHEN checking the availability of the ggshield auth flow web method on a dashboard of various
+        various, with or without the flow enabled
+        THEN it succeeds if the version is high enough, and the preference is enabled
+        """
+
+        def client_get_mock(*args, endpoint, **kwargs):
+            if endpoint == "metadata":
+                return Mock(
+                    ok=status_code < 400,
+                    status_code=status_code,
+                    json=lambda: {
+                        "version": version,
+                        "preferences": {
+                            "public_api__ggshield_auth_flow_enabled": preference_enabled
+                        }
+                        if preference_enabled is not None
+                        else {},
+                    },
+                )
+            raise NotImplementedError
```

Comment:
Should be `raise NotImplementedError()`, but maybe you can replace the `if endpoint == "metadata"` with `assert endpoint == "metadata"` instead?

Reply:
Oh nice, didn't know about raising the exception class.

I don't have strong opinions on `raise` vs `assert`, so LGTM.